src/change-data-capture/templates/deployment.yaml

```diff
@@ -59,13 +59,29 @@ spec:
               {{- toYaml .Values.securityContext | nindent 12 }}
           env:
             {{- include "harnesscommon.dbconnection.mongoEnv" . | indent 12 }}
+            {{- if .Values.global.database.timescaledb.installed }}
+            - name: TIMESCALEDB_USERNAME
+              value: postgres
             - name : TIMESCALEDB_PASSWORD
               valueFrom:
                 secretKeyRef:
                   name: {{ .Values.timescaleSecret.password.name }}
                   key: {{ .Values.timescaleSecret.password.key }}
-            - name: TIMESCALEDB_USERNAME
-              value: postgres
+            - name: TIMESCALEDB_URI
+              value: 'jdbc:postgresql://timescaledb-single-chart.{{ .Release.Namespace }}:5432/harness'
+            {{- else }}
+            {{ include "harnesscommon.dbconnection.dbenvuser" (dict "type" "timescale" "variableName" "TIMESCALEDB_USERNAME" "secret" .Values.global.database.timescaledb.secretName "userKey" .Values.global.database.timescaledb.userKey ) | nindent 12 }}
+            {{ include "harnesscommon.dbconnection.dbenvpassword" (dict "type" "timescale" "variableName" "TIMESCALEDB_PASSWORD" "secret" .Values.global.database.timescaledb.secretName "passwordKey" .Values.global.database.timescaledb.passwordKey ) | nindent 12 }}
```

Comment:
is there any way to unify username , secrets ?
could we use same field with default value for out of the box db ?

Reply:
the current external secret logic requires that customer provide the secrets to timescaledb and create the secret object.
Another issue if we make it common is that it might break the in-cluster timescale configurations